lib/fluent/plugin/formatter_csv.rb

```diff
@@ -37,6 +41,14 @@ def configure(conf)
         @fields = fields.select{|f| !f.empty? }
         raise ConfigError, "empty value is specified in fields parameter" if @fields.empty?
 
+        if @fields.any? { |f| record_accessor_nested?(f) }
+          @accessors = @fields.map { |f| record_accessor_create(f) }
```

Comment:
How about this? it can avoid the needless call of `Accessor#call`. Not nested fields don't need to be called this method, right?

```rb
nested = @fileds.select { |f| record_accessor_nested?(f) }
unless nested.empty?
  @accessors = nested.map { |f| record_accessor_create(f) }

   ...
end
```

Reply:
Your appoarch is bit complicated.
If fields are `a,$.b.c,d`, formatter_csv needs to remember the position when separate normal access and record_accessor.